lib/vox_web/controllers/restaurants_controller.ex

```diff
@@ -1,28 +1,42 @@
 defmodule VoxWeb.RestaurantsController do
+
   use VoxWeb, :controller
 
   alias Vox.Restaurants
   alias Vox.Restaurants.Restaurant
+  alias Vox.Votes
 
   def index(conn, _params) do
     restaurants = Restaurants.list_restaurants()
+    votes_by_restaurant = Votes.get_vote_count_for_restaurants()
     changeset = Restaurant.changeset(%Restaurant{}, %{})
-    render(conn, "index.html", restaurants: restaurants, changeset: changeset)
+    render(conn, "index.html", restaurants: restaurants, changeset: changeset, votes_by_restaurant: votes_by_restaurant)
   end
 
   def update(conn, %{"id" => id}) do
     restaurant = Restaurants.get_restaurant_by_id(id)
-    Restaurants.add_vote_to_restaurant(restaurant)
-    redirect(conn, to: Routes.restaurants_path(conn, :index))
+    user_id = conn.assigns.current_user.id
+    unless Votes.user_voted?(user_id) do
+      Votes.create_vote(%{user_id: user_id, restaurant_id: restaurant.id})
+    end
```

Comment:
this whole logic should be inside the context

Reply:
so create_vote calls the user_voted? instead doing it inside the controller

Reply:
I realised I didn't fix this yesterday, working on it

Reply:
ok this should do: